subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -2322,3 +2429,56 @@ int nrf_cloud_gnss_msg_json_encode(const struct nrf_cloud_gnss_data * const gnss
 
 	return ret;
 }
+
+int nrf_cloud_encode_alert(const struct nrf_cloud_alert_info *alert,
+			   struct nrf_cloud_data *output)
+{
+#if defined(CONFIG_NRF_CLOUD_ALERTS)
+	int ret;
+
+	__ASSERT_NO_MSG(alert != NULL);
+	__ASSERT_NO_MSG(output != NULL);
+
+	cJSON *root_obj = cJSON_CreateObject();
+
+	if (root_obj == NULL) {
+		return -ENOMEM;
+	}
+
+	ret = json_add_str_cs(root_obj, NRF_CLOUD_JSON_APPID_KEY, NRF_CLOUD_JSON_APPID_VAL_ALERT);
+	ret += json_add_num_cs(root_obj, NRF_CLOUD_JSON_ALERT_TYPE, alert->type);
+	ret += json_add_num_cs(root_obj, NRF_CLOUD_JSON_ALERT_VALUE, alert->value);
```

Comment:
I did not implement such a mechanism in order to keep the API simple.  I suppose I could define a special value for `value` that indicates "do not send", like `NAN`, rather than add a control parameter to say "do not send the value", or another API that does not include a value parameter.

Reply:
NAN seems pretty reasonable -- in fact, NAN cannot even be encoded into JSON, so it can't be sent anyways.

I do worry that someone's code may **_accidentally_** produce NAN as part of some kind of math bug, and then not find out because the data was silently ignored.

But I'm also not sure whether that edge-case is worth complicating the `nrf_cloud_alert_info` struct with a "do/do not send" flag. Instead we could just make sure to document that NAN will be treated as "do not send a value."

Reply:
I think the accidental case you mention can be helped by documenting this.  If their application normally always reports a value for a specific alert type, and they receive one without a value, the documentation will help them infer that the value is missing because it was a NAN generated by the value's data source.